packages/@jsii/spec/lib/replace-assembly.ts

```diff
@@ -0,0 +1,51 @@
+import * as crypto from 'crypto';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+import { Assembly } from './assembly';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
+const sortJson = require('sort-json');
+
+/**
+ * Options for replacing the assembly
+ */
+interface ReplaceAssemblyOptions {
+  /**
+   * The assembly file extension that you want to overwrite.
+   */
+  readonly fileExtension: string;
+
+  /**
+   * The directory where the assembly is located.
+   */
+  readonly directory: string;
+}
+
+/**
+ * Replaces the file where the original assembly file *should* be found with a new assembly file.
+ * Recalculates the fingerprint of the assembly to avoid tampering detection.
+ */
+export async function replaceAssembly(
+  assembly: Assembly,
+  options: ReplaceAssemblyOptions,
+): Promise<void> {
+  const fileName = path.join(options.directory, options.fileExtension);
```

Comment:
This is an open question in my mind -- should we hardcode `.jsii` or just expose it as an option for those that use `replaceAssembly`? I figure that this is more versatile but maybe unnecessary.

Reply:
How about we just pass a `fileName` to `saveAssembly()`, that saves us having to decide anything, leaves the flexibility, and makes for a still pretty sane API from the PoV of the consumer?